### PR TITLE
mapped token name

### PR DESCRIPTION
**Pull request by xiaoch05**

### CODE REVIEW

File: contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol [hunk 1]
```diff
@@ -88,14 +90,15 @@ contract DarwiniaMappingTokenFactory is Initializable, Ownable, DailyLimit {
...
+        address source,
+        string memory source_chain
     ) external onlySystem returns (address payable token) {
         bytes32 salt = keccak256(abi.encodePacked(backing, source));
         require(tokenMap[salt] == address(0), "contract has been deployed");
         bytes memory bytecode = type(TransparentUpgradeableProxy).creationCode;
         bytes memory erc20initdata = 
             abi.encodeWithSignature("initialize(string,string,uint8)",
-                                    name,
+                                    string(abi.encodePacked(name, "(", source_chain, "->", chain_name, ")")),
```
Comment (HackFisher) [inline contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol:101]:
```suggestion
                                    string(abi.encodePacked(name, "(", source_chain, ">", chain_name, ")")),
```
Please refer the latest changes in the doc.

File: contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol [hunk 2]
```diff
@@ -33,6 +33,7 @@ contract DarwiniaMappingTokenFactory is Initializable, Ownable, DailyLimit {
     mapping(address => TokenInfo) public tokenToInfo;
     mapping(string => address) public logic;
     mapping(bytes => UnconfirmedInfo) public transferUnconfirmed;
+    string public chain_name;
```
Comment (HackFisher) [inline contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol:36]:
```suggestion
    string public backing_chain_name;
```

File: contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol [hunk 3]
```diff
@@ -88,14 +90,15 @@ contract DarwiniaMappingTokenFactory is Initializable, Ownable, DailyLimit {
         string memory symbol,
         uint8 decimals,
         address backing,
-        address source
+        address source,
+        string memory source_chain
```
Comment (HackFisher) [inline contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol:94]:
```suggestion
        string memory issuing_chain_name
```

**Review by HackFisher — CHANGES_REQUESTED** [review]
Please update using precise naming according to the docs.

File: contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol [hunk 4]
```diff
@@ -88,14 +90,15 @@ contract DarwiniaMappingTokenFactory is Initializable, Ownable, DailyLimit {
         string memory symbol,
         uint8 decimals,
         address backing,
-        address source
+        address source,
```
Comment (HackFisher) [inline contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol:93]: What's the meaning of `backing` and `source`?

Comment (xiaoch05) [reply contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol:93]: backing is the address of the backing model(backing contract address), and source is the origin erc20 token address.

Comment (xiaoch05) [reply contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol:93]: maybe it's better to call it backing_address and origin_address?

Comment (HackFisher) [reply contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol:93]: How about `backing_contract` and `original_token`?

`original_token` means the token address on backing chain being mapped, and `mapping_token` means the mapping token generated on issuing chain, relative to `original_token`.

Comment (xiaoch05) [reply contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol]: `original_token` and `mapping_token` seems good.

**Review by HackFisher — APPROVED** [review]

**Pull request merged**